### PR TITLE
feat(amplify-codegen-appsync-model-plugin): java tostring (#3772)

### PR DESCRIPTION
**Pull request by jamesonwilliams**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class AppSyncModelJavaVisitor<
     const result: string[] = [this.generatePackageName(), '', this.generateImportStatements(LOADER_IMPORT_PACKAGES)];
     result.push(
       transformComment(dedent` Contains the set of model classes that implement {@link Model}
-    interface.`)
+    interface.`),
```
Comment (jamesonwilliams) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]: **These get automatically added by a pre-commit hook.**

File: packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts [hunk 2]
```diff
@@ -601,14 +605,33 @@ export class AppSyncModelJavaVisitor<
...
+      `.append("${this.getModelName(model)} {")`,
+      ...this.getNonConnectedField(model).map((field, index) => {
+        const fieldName = this.getFieldName(field);
+        const fieldGetterName = this.getFieldGetterName(field);
+        if (index == 0) {
+          return '.append("' + fieldName + '=" + String.valueOf(' + fieldGetterName + '()))';
+        } else {
+          return '.append(" , ' + fieldName + '=" + String.valueOf(' + fieldGetterName + '()))';
+        }
+      }),
```
Comment (yuth) [inline packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-java-visitor.ts]:
```suggestion
      ...this.getNonConnectedField(model).map((field, index) => {
        const fieldName = this.getFieldName(field);
        const fieldGetterName = this.getFieldGetterName(field);

          return '.append("' + fieldName + '=" + String.valueOf(' + fieldGetterName + '()))';
      }).join(',\n'),
```

**Comment by jamesonwilliams** [comment]
> Code wise changes look good. I see a few snapshot tests failing. Could you update those and then we can merge this change.

I may need your help with this! I couldn't figure out what's going wrong with the snapshot.

**Comment by jamesonwilliams** [comment]
I'm not sure how to complete this work.

I've opened https://github.com/aws-amplify/amplify-cli/issues/3870 to ask for updates to the Contributor's Guide.

**Pull request merged**